### PR TITLE
cmake: check for unaligned access

### PR DESCRIPTION
**Pull request by tchaikov**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/crimson/net/ProtocolV1.cc [hunk 1]
```diff
@@ -25,6 +25,26 @@
...
+struct unaligned_t;
+
+template<typename T>
+struct unaligned_t<T, enable_if_t<std::is_pod_v<T>>>
+{
+  T value;
+} __attribute__((packed));
+
+template<typename T>
+inline T unaligned_load(const char* p)
```
Comment (cyx1231st) [inline src/crimson/net/ProtocolV1.cc]: Does it also applies to `ProtocolV2.cc`? such as:
* https://github.com/ceph/ceph/blob/master/src/crimson/net/ProtocolV2.cc#L207
* https://github.com/ceph/ceph/blob/master/src/crimson/net/ProtocolV2.cc#L289

Comment (tchaikov) [reply src/crimson/net/ProtocolV1.cc]: sure. i was testing with v1 protocol. will fix v2 as well in the next revision.

Comment (tchaikov) [reply src/crimson/net/ProtocolV1.cc]: @cyx1231st i withdraw my words. the accessed structs in v2 are packed. so they are immune to the unaligned access issue.

**Comment by tchaikov** [comment]
chagelog:

 - rebased to latest master
 - fix the typos in commit message

**Comment by tchaikov** [comment]
will run this change through a rados bench before merging it, just to make sure it does not introduce performance regression.

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

File: src/crimson/net/ProtocolV1.cc [hunk 2]
```diff
@@ -25,6 +25,26 @@
...
+{
+  using unaligned = unaligned_t<T>;
```
Comment (cbodley) [inline src/crimson/net/ProtocolV1.cc]: can simplify out the partial specialization of unaligned_t:
```c++
template <typename T>
inline T unaligned_load(const char* p)
{
  static_assert(std::is_pod_v<T>);
  struct unaligned { T value; } __attribute__((packed));
  ...
```

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

File: src/crimson/net/ProtocolV1.cc (hunk 2 above)
Comment (tchaikov) [reply src/crimson/net/ProtocolV1.cc]: @cbodley thanks for the suggestion! i have a better understanding of SFINAE now.

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
```
[ceph-perf-pull-requests] $ /bin/sh -xe /tmp/jenkins8914535021595273041.sh
+ /home/jenkins-build/build/workspace/ceph-perf-pull-requests/cbt/compare.py -a /home/jenkins-build/build/workspace/ceph-perf-pull-requests/cbt-results/pr -b /home/jenkins-build/build/workspace/ceph-perf-pull-requests/cbt-results/master -v
03:47:47 - DEBUG    - cbt      - Nodes : incerta08.front.sepia.ceph.com
03:47:47 - DEBUG    - cbt      - Nodes : incerta08.front.sepia.ceph.com
03:47:47 - DEBUG    - cbt      - Nodes : incerta08.front.sepia.ceph.com
03:47:47 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/0: bandwidth: (or (greater) (near 0.05)):: 0.151126/0.261884  => rejected
03:47:47 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/0: iops_avg: (or (greater) (near 0.05)):: 38.0/67.0  => rejected
03:47:47 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/0: iops_stddev: (or (less) (near 0.05)):: 13.4536/11.2694  => rejected
03:47:47 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/0: latency_avg: (or (less) (near 0.05)):: 0.412265/0.235164  => rejected
03:47:47 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/1: bandwidth: (or (greater) (near 0.05)):: 0.140227/0.249925  => rejected
03:47:47 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/1: iops_avg: (or (greater) (near 0.05)):: 35.0/63.0  => rejected
03:47:47 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/1: iops_stddev: (or (less) (near 0.05)):: 10.8167/12.741  => accepted
03:47:47 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/1: latency_avg: (or (less) (near 0.05)):: 0.440548/0.249323  => rejected
03:47:47 - WARNING  - cbt      - seq/incerta08.front.sepia.ceph.com/0: bandwidth: (or (greater) (near 0.05)):: 0.928827/0.988699  => rejected
03:47:47 - WARNING  - cbt      - seq/incerta08.front.sepia.ceph.com/0: iops_avg: (or (greater) (near 0.05)):: 237.0/253.0  => rejected
03:47:47 - INFO     - cbt      - seq/incerta08.front.sepia.ceph.com/0: iops_stddev: (or (less) (near 0.05)):: 0.0/0.0  => accepted
03:47:47 - WARNING  - cbt      - seq/incerta08.front.sepia.ceph.com/0: latency_avg: (or (less) (near 0.05)):: 0.0670771/0.06279  => rejected
03:47:47 - WARNING  - cbt      - seq/incerta08.front.sepia.ceph.com/1: bandwidth: (or (greater) (near 0.05)):: 0.81442/1.00633  => rejected
03:47:47 - WARNING  - cbt      - seq/incerta08.front.sepia.ceph.com/1: iops_avg: (or (greater) (near 0.05)):: 208.0/257.0  => rejected
03:47:47 - INFO     - cbt      - seq/incerta08.front.sepia.ceph.com/1: iops_stddev: (or (less) (near 0.05)):: 0.0/0.0  => accepted
03:47:47 - WARNING  - cbt      - seq/incerta08.front.sepia.ceph.com/1: latency_avg: (or (less) (near 0.05)):: 0.0765027/0.0615222  => rejected
03:47:47 - WARNING  - cbt      - 13 tests failed out of 16
```

i am removing the crimson change from this PR.

**Pull request merged**